pkg/globalinit/global_initializers_test.go

```diff
@@ -0,0 +1,211 @@
+			expected:     true,
+		},
+	}
+	for _, spec := range tests {
+		t.Run(spec.test, func(t *testing.T) {
+			assert := assert.New(t)
+
+			// Reset any previous initializers
+			initializers = nil
+
+			for _, triggerFunc := range spec.triggerFuncs {
+				RegisterInitializer(triggerFunc, func(io.Writer) error { return nil })
+			}
+
+			assert.Equal(spec.expected, InitializationRequired())
+		})
+	}
+}
+
+func TestPerformInitializations(t *testing.T) {
+	tests := []struct {
+		test          string
+		triggerFuncs  []func() bool
+		initFuncs     []func(io.Writer) error
+		expectError   bool
+		expectedOut   []string
+		unexpectedOut []string
+	}{
+		{
+			test: "single true trigger",
+			triggerFuncs: []func() bool{
+				func() bool { return true },
+			},
+			initFuncs: []func(io.Writer) error{
+				func(w io.Writer) error { fmt.Fprintln(w, "1"); return nil },
+			},
+			expectError:   false,
+			expectedOut:   []string{"1"},
+			unexpectedOut: []string{"2", "3"},
+		},
+		{
+			test: "single false trigger",
+			triggerFuncs: []func() bool{
+				func() bool { return false },
+			},
+			initFuncs: []func(io.Writer) error{
+				func(w io.Writer) error { fmt.Fprintln(w, "1"); return nil },
+			},
+			expectError:   true,
+			unexpectedOut: []string{"1", "2", "3"},
+		},
+		{
+			test: "multiple true triggers",
+			triggerFuncs: []func() bool{
+				func() bool { return true },
+				func() bool { return true },
+				func() bool { return true },
+			},
+			initFuncs: []func(io.Writer) error{
+				func(w io.Writer) error { fmt.Fprintln(w, "1"); return nil },
+				func(w io.Writer) error { fmt.Fprintln(w, "2"); return nil },
+				func(w io.Writer) error { fmt.Fprintln(w, "3"); return nil },
+			},
+			expectError: false,
+			expectedOut: []string{"1", "2", "3"},
+		},
+		{
+			test: "multiple false triggers",
+			triggerFuncs: []func() bool{
+				func() bool { return false },
+				func() bool { return false },
+				func() bool { return false },
+			},
+			initFuncs: []func(io.Writer) error{
+				func(w io.Writer) error { fmt.Fprintln(w, "1"); return nil },
+				func(w io.Writer) error { fmt.Fprintln(w, "2"); return nil },
+				func(w io.Writer) error { fmt.Fprintln(w, "3"); return nil },
+			},
+			expectError:   true,
+			unexpectedOut: []string{"1", "2", "3"},
+		},
+		{
+			test: "mix of true and false triggers",
+			triggerFuncs: []func() bool{
+				func() bool { return false },
+				func() bool { return true },
+				func() bool { return false },
+			},
+			initFuncs: []func(io.Writer) error{
+				func(w io.Writer) error { fmt.Fprintln(w, "1"); return nil },
+				func(w io.Writer) error { fmt.Fprintln(w, "2"); return nil },
+				func(w io.Writer) error { fmt.Fprintln(w, "3"); return nil },
+			},
+			expectError:   false,
+			expectedOut:   []string{"2"},
+			unexpectedOut: []string{"1", "3"},
+		},
+		{
+			test: "init throws error",
```

Comment:
nit: I think one interesting case to include in multiple initFuncs, returning a mix of success and error (e.g. erro just on the second of 3). Demonstrates that error does not interrupt other initializers.